public/firebase-messaging-sw.js

```diff
@@ -0,0 +1,43 @@
+// [START initialize_firebase_in_sw]
+// Give the service worker access to Firebase Messaging.
+// Note that you can only use Firebase Messaging here, other Firebase libraries
+// are not available in the service worker.
+importScripts('https://www.gstatic.com/firebasejs/7.12.0/firebase-app.js');
+importScripts('https://www.gstatic.com/firebasejs/7.12.0/firebase-messaging.js');
+
+// Initialize the Firebase app in the service worker by passing in
+// your app's Firebase config object.
+// https://firebase.google.com/docs/web/setup#config-object
+firebase.initializeApp({
```

Comment:
Is there a reason not to use the firebaseconfig.json here as well?

Reply:
can't use `require` for importing files as it's part of a publicly served folder `public`

Reply:
maybe use webpack to process it? feels off having this data twice

Reply:
Can you help how this is done? It's public data anyway tho.

Reply:
not about being public but about not having it in two places

Reply:
make the whole JSON a string and put in env variable?

Reply:
and then use webpack to replace the env variable with the actual value

Reply:
ok let's leave it for now